subprojects/dependency-management/src/integTest/groovy/org/gradle/integtests/resolve/transform/ArtifactTransformCachingIntegrationTest.groovy

```diff
@@ -1058,6 +1059,108 @@ class ArtifactTransformCachingIntegrationTest extends AbstractHttpDependencyReso
         gcFile.lastModified() >= SECONDS.toMillis(beforeCleanup)
     }
 
+    def "cache cleanup does not delete entries that are currently being created"() {
+        given:
+        requireOwnGradleUserHomeDir() // needs its own journal
+
+        and:
+        def transformBarrier = file("transform-barrier").touch()
+        def cleaningBarrier = file("cleaning-barrier").touch()
+        buildFile << declareAttributes() << """
+            class BlockingTransform extends ArtifactTransform {
+                List<File> transform(File input) {
+                    def output = new File(outputDirectory, "\${input.name}.txt");
+                    output.text = ""
+                    println "Transformed \$input.name to \$output.name into \$outputDirectory"
+                    def barrier = new File('${transformBarrier.absolutePath}')
```

Comment:
We use `BlockingHttpServer` in other places to synchronize actions in tests. Can you use it here maybe, too?

Reply:
So much to learn... 🙇

Done in 598682be677.